x/ibc-transfer/keeper/relay.go

```diff
@@ -12,19 +10,42 @@ import (
 
 // SendTransfer handles transfer sending logic. There are 2 possible cases:
 //
-// 1. Sender chain is the source chain of the coins (i.e where they were minted): the coins
-// are transferred to an escrow address (i.e locked) on the sender chain and then
-// transferred to the destination chain (i.e not the source chain) via a packet
-// with the corresponding fungible token data.
+// 1. Sender chain is acting as the source zone. The coins are transferred
+// to an escrow address (i.e locked) on the sender chain and then transferred
+// to the receiving chain through IBC TAO logic. It is expected that the
+// receiving chain will mint vouchers to the receiving address.
+//
+// 2. Sender chain is acting as the sink zone. The coins (vouchers) are burned
+// on the sender chain and then transferred to the receiving chain though IBC
+// TAO logic. It is expected that the receiving chain, which had previously
+// sent the original denomination, will unescrow the fungible token and send
+// it to the receiving address.
+//
+// Another way of thinking of source and sink zones is through the token's
+// timeline. Each send to any chain other than the one it was previously
+// received from is a movement forwards in the token's timeline. This causes
+// trace to be added to the token's history and the destination port and
+// destination channel to be prefixed to the denomination. In these instances
+// the sender chain is acting as the source zone. When the token is sent back
+// to the chain it previously received from, the prefix is removed. This is
+// a backwards movement in the token's timeline and the sender chain
+// is acting as the sink zone.
+//
+// Example:
```

Comment:
Could you also describe tokens moving in the reverse direction in this example

Reply:
Hmm that is what A -> C covers? did you want me to add all the way back to origin?

Reply:
updated